applications/DEMApplication/python_scripts/DEM_analysis_stage.py

```diff
@@ -451,7 +451,7 @@ def ConvertClusterFileNamesFromRelativePathToAbsolutePath(self):
 
     def RunAnalytics(self, time, is_time_to_print=True):
         self.MakeAnalyticsMeasurements()
-        if is_time_to_print:
+        if is_time_to_print and self.FaceAnalyzerClass.ghost_smp_detected:
```

Comment:
Isn't it possible to put the second part of the` if` inside the function `MakeAnalyticsPipeLine `? So that you don't need to access a member variable from outside...